src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.impl.OperationStrategyImpl;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy strategy;
+    private HashMap storage;
+
+    public ShopServiceImpl() {
+        strategy = new OperationStrategyImpl();
+        storage = new HashMap<String, Integer>();
```

Comment:
The `OperationStrategy` should not be instantiated directly within the `ShopServiceImpl` class. Instead, it should be passed as a parameter to the constructor, making the service more flexible and adhering to the Dependency Inversion principle of SOLID. Consider using Dependency Injection to provide an instance of `OperationStrategy`.